common/views/components/GridViewerPrototype/GridViewerPrototype.tsx

```diff
@@ -0,0 +1,195 @@
+import styled from 'styled-components';
+import {
+  useState,
+  memo,
+  useEffect,
+  useRef,
+  useContext,
+  RefObject,
+  FunctionComponent,
+  CSSProperties,
+} from 'react';
+import { FixedSizeGrid, FixedSizeList, areEqual } from 'react-window';
+import useScrollVelocity from '@weco/common/hooks/useScrollVelocity';
+import LL from '@weco/common/views/components/styled/LL';
+import IIIFCanvasThumbnail from '../IIIFCanvasThumbnailPrototype/IIIFCanvasThumbnailPrototype';
+import Space from '@weco/common/views/components/styled/Space';
+import GlobalInfoBarContext from '@weco/common/views/components/GlobalInfoBarContext/GlobalInfoBarContext';
+import { IIIFCanvas } from '@weco/common/model/iiif';
+import ItemViewerContext from '../ItemViewerContext/ItemViewerContext';
+
+const ThumbnailSpacer = styled(Space).attrs({
+  v: { size: 's', properties: ['padding-top', 'padding-bottom'] },
+})`
+  height: 200px;
+`;
+
+type CellProps = {
+  style: CSSProperties;
+  columnIndex: number;
+  rowIndex: number;
+  index: number;
+  data: {
+    scrollVelocity: number;
+    columnCount: number;
+    gridVisible: boolean;
+    setGridVisible: (value: boolean) => void;
+    mainViewerRef: RefObject<FixedSizeList>;
+    activeIndex: number;
+    setActiveIndex: (i: number) => void;
+    canvases: IIIFCanvas[];
+  };
+};
+
+const Cell = memo(({ columnIndex, rowIndex, style, data }: CellProps) => {
+  const {
+    columnCount,
+    mainViewerRef,
+    gridVisible,
+    setGridVisible,
+    scrollVelocity,
+    activeIndex,
+    setActiveIndex,
+    canvases,
+  } = data;
+  const itemIndex = rowIndex * columnCount + columnIndex;
+  const currentCanvas = canvases[itemIndex];
+  return (
+    <div style={style}>
+      {scrollVelocity > 1 ? (
+        <div style={{ display: 'flex', justifyContent: 'center' }}>
+          <LL lighten={true} />
+        </div>
+      ) : (
+        currentCanvas && (
+          <ThumbnailSpacer>
+            <IIIFCanvasThumbnail
+              canvas={currentCanvas}
+              clickHandler={() => {
+                mainViewerRef &&
+                  mainViewerRef.current &&
+                  mainViewerRef.current.scrollToItem(itemIndex);
+                setActiveIndex(itemIndex);
+                setGridVisible(false);
+              }}
+              isActive={activeIndex === itemIndex}
+              thumbNumber={itemIndex + 1}
+              isFocusable={gridVisible}
+            />
+          </ThumbnailSpacer>
+        )
+      )}
+    </div>
+  );
+}, areEqual);
+
+Cell.displayName = 'Cell';
+
+type GridViewerElProps = {
```

Comment:
We use `Container` elsewhere no?

Reply:
Do you mean as a naming convention? This is largely copy-pasted from the existing `GridViewer` component, but the name refers to the fact it contains a grid of thumbnails, rather than saying anything about its own intrinsic grid properties. Or did you mean something else?

Reply:
Yea - do you have any tips on how to review this as I am not sure if I am looking at some copy/pasted stuff (which could probably do with some work) or new stuff?